pkg/k8s/volumes/crud.go

```diff
@@ -155,6 +155,25 @@ func checkPVCValues(pvc *apiv1.PersistentVolumeClaim, dev *model.Dev, devPath st
 			)
 		}
 	}
+
+	if len(pvc.Spec.AccessModes) == 1 && dev.PersistentVolumeAccessMode() != pvc.Spec.AccessModes[0] {
```

Comment:
This force a failure if the user modifies the access modes or volume mode of an existing PVC